File: Itido-BMS/RentAPI/Rent/Models/TimePlanning/AbsenceReason.cs
using System;
using System.Collections.Generic;
using System.Linq.Expressions;

namespace Rent.Models.TimePlanning

{

    public class AbsenceReason
    {
        public int ID { get; set; }
        public string Description { get; set; }
        public bool CanUserCreate { get; set; }
        public bool CanUserRequest { get; set; }
        public bool CanManagerCreate { get; set; }
        public bool CanManagerRequest { get; set; }
        public virtual ICollection<Absence> Absence { get; set; }

        public static Expression<Func<AbsenceReason, dynamic>> BasicDTO()
        {
            return a => a != null ?
            new
            {
                a.ID,
                a.Description,
            } : null;
        }
    }
}

File: Itido-BMS/dart/bms_dart/lib/src/models/absence_reason.dart
class AbsenceReason {
  int id;
  String description;
  bool canUserRequest;
  bool canUserCreate;
  bool canManagerRequest;
  bool canManagerCreate;

  AbsenceReason({
    this.id,
    this.description,
    this.canUserRequest,
    this.canUserCreate,
    this.canManagerRequest,
    this.canManagerCreate,
  });

  factory AbsenceReason.fromJson(json) {
    if (json == null) return null;
    return AbsenceReason._fromJson(json);
  }

  AbsenceReason._fromJson(json)
      : this.id = json['id'],
        this.description = json['description'],
        this.canUserRequest = json['canUserRequest'],
        this.canUserCreate = json['canUserCreate'],
        this.canManagerRequest = json['canManagerRequest'],
        this.canManagerCreate = json['canManagerCreate'];

  Map<String, dynamic> toMap() => {
        'id': this.id,
        'description': this.description,
        'canUserRequest': this.canUserRequest,
        'canUserCreate': this.canUserCreate,
        'canManagerRequest': this.canManagerRequest,
        'canManagerCreate': this.canManagerCreate,
      };
}
